bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
So far we have avoided the word "device" everywhere - it isn't part of the ESH vocabulary. From the JavaDoc here, the right word in the ESH context would imho be "thing".

I think you used "device" here, because you felt that a new method "getThingId()" would only cause irritation - I agree.
But wouldn't the best solution to this be to go with the vocabulary, which we already introduced for the DiscoveryResult? I.e. could we simply add a new method `getRepresentationProperty()`, which gives the handler the chance to point to the property to use for unique identification?